### PR TITLE
Flow routing speed-up with Numba (5-10X faster)

### PR DESCRIPTION
**Pull request by irbreeves**
I put the overwash routing for an individual storm timestep into its own function and then use Numba to speed it up. Numba only works for functions with static parameters (can't use `self`). This change makes the model 5-10 times faster; the run_B3D_BMI script was taking ~25 seconds with the original code on my machine, but now takes ~3 seconds with this change.

### CODE REVIEW

File: barrier3d/barrier3d.py [hunk 1]
```diff
@@ -779,6 +756,413 @@ def migrate_dunes(
...
+                       Discharge,
+                       Rin,
+                       Elevation,
+                       SedFluxIn,
+                       SedFluxOut,
+                       ShrubDomainWidth,
+                       DeadDomainWidth,
+                       inundation,
+                       C):
+
```
Comment (anardek) [inline barrier3d/barrier3d.py:793]: Can you add a single comment that says what this does?

File: barrier3d/barrier3d.py [hunk 2]
```diff
@@ -779,6 +756,413 @@ def migrate_dunes(
                 self._SCRagg[time_index] + cellular_shoreline_change
             )  # Reset (make more positive), leaving residual
 
+    @staticmethod
```
Comment (anardek) [inline barrier3d/barrier3d.py:759]: can you add a comment here that says briefly what this does (basically the name of this pull request)

**Review by anardek — COMMENTED** [review]
baller

File: barrier3d/barrier3d.py (hunk 1 above)
Comment (irbreeves) [reply barrier3d/barrier3d.py:795]: Done, see new commit

File: barrier3d/barrier3d.py (hunk 2 above)
Comment (irbreeves) [reply barrier3d/barrier3d.py]: Done, see new commit

**Comment by anardek** [comment]
@avanblunk and @franklin1895, I would pull this most recent version into your branches

**Pull request merged**